src/Domain/LeanCode.DomainModels.EF/EFRepository.cs

```diff
@@ -58,4 +60,28 @@ public virtual void Update(TEntity entity)
     }
 
     public abstract Task<TEntity?> FindAsync(TIdentity id, CancellationToken cancellationToken = default);
+
+    [SuppressMessage(
+        "?",
+        "EF1001",
+        Justification = "There basically resembles the EntityFinder.FindTracked to mimic `FindAsync` snapshot re-use."
+    )]
+    protected TEntity? FindTracked(TIdentity id)
+    {
+        // Safety: aggregates are bound to have Id as a primary key by design.
+        var primaryKey = DbContext.Model.FindEntityType(typeof(TEntity))!.FindPrimaryKey()!;
+        return ((IDbContextDependencies)DbContext).StateManager!.TryGetEntryTyped(primaryKey, id)?.Entity as TEntity;
+    }
+
+    protected ValueTask<TEntity?> FindTrackedOrLoadNewAsync(
+        TIdentity id,
+        IQueryable<TEntity> withIncludes,
```

Comment:
I find it a bit awkward that I can essentially put any queryable here, while in most cases I'm going to use plain `DbSet` (we tend to do owned types anyway plus there's a `Navigation().AutoInclude()` option).

Maybe 
```
    protected ValueTask<TEntity?> FindTrackedOrLoadNewAsync(
        TIdentity id,
        CancellationToken cancellationToken
        Func<IQueryable<TEntity>,IQueryable<TEntity>>? includes = null)
``` 
Placement of cancellation token is a bit awkward here, but in this context it's better than having two overloads, one with `includes` and one without.

Reply:
This "tend" with owned types is, IMO, exaggerated. It's not that common for sub-entities. Also, if you use owned types **and** the db context is tracking, then you don't need this mumbo jumbo - `FindAsync` will work (as this is basically FindAsync reimplemented).

We don't work on dbsets, unfortunately - when you do anything with it (include or AsTracking), you end up with IQueryable.

And, to be honest, having lambda instead of plain object is not that much different for me. I can change that (except the cancelation token, let's not deviate from the standard)

Reply:
Random thought, but considering

> It is very important that the `Include`s used in all calls stay the same (otherwise, you might get entity that looks different by mistake)

… maybe we could take this opportunity to rework this class a bit and instead of having the consumer/subclasses implement `FindAsync`, they could instead implement `protected abstract IQueryable<TEntity> QueryWithIncludes();` (or `PrepareQuery`) with the default implementation of `dbSet.AsQueryable()`?

Then we could use this new method to feed the same include chain to both `Find` methods' default implementations.

Reply:
Did that, plus changed how the method looks like to better support composite keys (thus - arrays).